x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/model/rule_assets/prebuilt_rule_asset.ts

```diff
@@ -76,7 +55,7 @@ function zodMaskFor<T>() {
  */
 export type PrebuiltRuleAsset = z.infer<typeof PrebuiltRuleAsset>;
 export const PrebuiltRuleAsset = BaseCreateProps.omit(BASE_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET)
-  .and(TypeSpecificFields)
```

Comment:
@jpdjere 👋 Hey, could you take a look at this one please? I removed the changes for `TypeSpecificFields` because response_actions are added to optionalFields to all rule types. 
However wanted to hear from you - because not totally sure what were the intentions of this. 
Thanks!

Reply:
Hi @tomsonpl. Yes, if `response_actions` is now going to be part of the base optional fields for all rules types, your changes make sense: what we were doing in this Zod transform was removing `response_actions` manually from each of the rule types where it was present. That won't be needed anymore.

**However:** see above where we remove the unwanted fields from the base props. We should add `response_actions` to the list there now, in order to remove it:
```ts
const BASE_PROPS_REMOVED_FROM_PREBUILT_RULE_ASSET = zodMaskFor<BaseCreateProps>()([
  'actions',
  'response_actions', // this is new
  'throttle',
  'meta',
  'output_index',
  'namespace',
  'alias_purpose',
  'alias_target_id',
  'outcome',
]);
```
---
But also, one question in general: the description in this PR says that we're adding `response_actions` to ESQL, EQL and NewTerms rules. But by adding `response_actions`  to `BaseOptionalFields` in asdfadfas, we're actually adding it to the schema of all rule types (including Threshold, Machine Learning and Threat Intel). Is this correct/expected?

Reply:
@tomsonpl I just merged a PR that changed this file and created a conflict here. However, the idea described above is the same. Let me know if you need some help fixing the conflict.

Reply:
Hey, I think that somewhere before it was suggested to add it to `BaseOptionalFields`, even though it doesn't reflect all the rules. Do you think I shouldn't do so? You got it right that it only affects Query, ESQL, EQL, NewTerms.

Reply:
I ended up removing the `omit` functionality since it was only affecting `response_actions`. Let me know if you'd prefer to leave the functionality with empty array or something. 
Thanks!

Reply:
> Hey, I think that somewhere before it was suggested to add it to BaseOptionalFields, even though it doesn't reflect all the rules. Do you think I shouldn't do so? You got it right that it only affects Query, ESQL, EQL, NewTerms.

I think it should be OK, but I'm not sure if you'll end up having to do a pretty huge refactor to satisfy TS.

> I ended up removing the omit functionality since it was only affecting response_actions. Let me know if you'd prefer to leave the functionality with empty array or something.

Left an additional comment: https://github.com/elastic/kibana/pull/191874/files#r1757569972